admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/presentation/screen/dukan_requests/DukanRequestsViewModel.kt

```diff
@@ -105,14 +113,87 @@ class DukanRequestsViewModel(
         getRequestedDukans()
     }
 
-    override fun onApproveDukanClicked(dukanId: Uuid) {}
+    override fun onApproveDukanClicked() {
+        tryToExecute(
+            callee = {
+                dukanRepository.updateDukanStatus(
+                    dukanId = currentState.selectedDukanId!!,
+                    status = Dukan.Status.APPROVED,
+                    message = currentState.rejectReason
+                )
+            },
+            onSuccess = {
+                onDismissDukanDetails()
+                getRequestedDukans()
+                viewModelScope.launch {
+                    showSnackBar(
+                        title = stringProvider.getString(Res.string.status_updated_title),
+                        message = stringProvider.getString(Res.string.dukan_approved_successfully),
+                        isSuccess = true
+                    )
+                }
+            },
```

Comment:
I think it will be good to make function `onSuccessDukanApproved`
like

```suggestion
            onSuccess = { onSuccessDukanApproved()},
            onError = ::onError,
            dispatcher = dispatcher
            )
            }
            private fun onSuccessDukanApproved(){
            onDismissDukanDetails()
                getRequestedDukans()
                viewModelScope.launch {
                    showSnackBar(
                        title = stringProvider.getString(Res.string.status_updated_title),
                        message = stringProvider.getString(Res.string.dukan_approved_successfully),
                        isSuccess = true
                    )
                }
            }
```

Reply:
ok great